clients/src/main/java/org/apache/kafka/clients/consumer/internals/AbstractCoordinator.java

```diff
@@ -888,9 +888,12 @@ public void run() {
                         long now = time.milliseconds();
 
                         if (coordinatorUnknown()) {
-                            if (findCoordinatorFuture == null)
+                            if (findCoordinatorFuture == null) {
                                 lookupCoordinator();
-                            else
+                                // coordinator may still be unknown, instead of an immediate retry, let's wait for a bit.
+                                if (coordinatorUnknown())
```

Comment:
Are we attempting to handle the case that there are no brokers available to send the GroupCoordinator request to?